jstz_cli/src/repl.rs

```diff
@@ -16,7 +16,7 @@ use crate::config::Config;
 pub fn exec(self_address: Option<String>, cfg: &Config) -> Result<()> {
     let address = &cfg.accounts.account_or_current(self_address)?.address;
 
-    let mut rt = Runtime::new().expect("Failed to create a new runtime.");
+    let mut rt = Runtime::new(usize::MAX).expect("Failed to create a new runtime.");
```

Comment:
It would be more ideal to split the runtime interface into creating a runtime w/out gas and one with. 
e.g. 
```rust
#[derive(Default)]
pub struct RuntimeOptions {
  gas_limit: Option<u32>
}

impl Runtime {
  pub fn new(options: &RuntimeOptions) -> JsResult<Self> {
     ...
  }  
}
```

Reply:
I'd disagree. 
1. The struct provides a `Default` implementation for default options
2. I think it would be possible to max-out the repl by running extremely large computations on the REPL -- moreover, conceptually it makes sense for gas to be an optional part of our runtime

Reply:
> Gas is not an optional part of the rollup.

Yes, but our runtime needs to be designed to also work in a local environment (this is absolutely key)

> None is not a sensible default, it's the wrong type. The runtime requires a usize and there isn't a sensible default.

`None` is a sensible default, since the `instructions_remaining` limit can optionally be passed to the context builder. 

Additionally the `Option` provides a clearer interface, clear semantics (is metering enabled or not) & better engineering practices (avoiding max integer defaults).

Reply:
> Yes, but our runtime needs to be designed to also work in a local environment (this is absolutely key)
This would be worth discussing, but that is a compile time decision not a runtime decision.  It would be possible to have this as a feature flag, but we can introduce this later.

> None is a sensible default, since the instructions_remaining limit can optionally be passed to the context builder.
No.  It's not an optional parameter, the boa developers are using the builder pattern which requires a default value, but not a sensible one.  If no value is supplied instructions remaining will be 0.

> Additionally the Option provides a clearer interface, clear semantics (is metering enabled or not) & better engineering practices (avoiding max integer defaults).
No. There is no default value, it is not an optional parameter.  There are special circumstances (the repl) under which we pass `usize::max` which is effectively unlimited gas.  The parameter is optional for now in the CLI to make it more convenient for us to test things.  Passing `usize::max` makes the intention far clearer than passing `None`.

Reply:
> This would be worth discussing, but that is a compile time decision not a runtime decision. It would be possible to have this as a feature flag, but we can introduce this later.

Can we introduce it now? Feature flags aren't complicated + will likely pay off in the future since `jstz_core` is likely to have feature flags as we try to improve crate separation & interfaces between them.

> No. It's not an optional parameter, the boa developers are using the builder pattern which requires a default value, but not a sensible one. If no value is supplied instructions remaining will be 0.

Ah I was unaware of this, I assumed they'd pick a better mechanism for this. 

> Passing usize::max makes the intention far clearer than passing None.

I'd disagree with this, passing `None` makes far more sense when you don't want to have a limit. Almost every metered runtime I know (including the Michelson interpreter in Tezos) provide an interface for optional gas.

Reply:
I think we've now strayed far enough away from the original point that I can consider this matter to be closed.